hardware/P1MeterBase.cpp

```diff
@@ -159,26 +159,29 @@ bool P1MeterBase::MatchLine()
 				l_exclmarkfound=1;
 				found=1;
 			}
-			else 
+			else
 				continue;
 			break;
 		default:
 			continue;
 		} //switch
-		
+
 		if(!found)
 			continue;
 
 		if (l_exclmarkfound) {
 			time_t atime=mytime(NULL);
-			sDecodeRXMessage(this, (const unsigned char *)&m_p1power, "Power", 255);
 			bool bSend2Shared=(difftime(atime,m_lastSharedSendElectra)>59);
-			if (std::abs(double(m_lastelectrausage)-double(m_p1power.usagecurrent))>40)
-				bSend2Shared=true;
+			if (
+				(std::abs(double(m_lastelectrausage)-double(m_p1power.usagecurrent))>40)&&
+				(difftime(atime,m_lastSharedSendElectra)>29)
```

Comment:
I thought to keep the original logic to always send every 60 seconds, wasn't sure why this is included (some form of heartbeat?).

It could be changed to 10 seconds, but is it really necessary to update that often? I thought that 30 secs (or even 60 secs) would be fine in every case.

I realize the data is processed every second, but that doesn't seem to cause much load. The update routine on the other hand is beating on my little RPi's CPU ;-) Could be worse in my case because I'm running the DB on a NFS share.

The (average) load difference is about 0.3, but especially the peak load has decreased significantly from 1.4 to 0.7 (CPU freq between 700 and 1000MHz).